mysql-test/mysql-test-run.pl

```diff
@@ -741,8 +739,7 @@ ($$$)
               }
 	      return ("Failure", 1, $completed, $extra_warnings);
 	    }
-	    elsif ($opt_max_test_fail > 0 and
-		   $num_failed_test >= $opt_max_test_fail) {
+	    elsif ($num_failed_test >= $opt_max_test_fail) {
```

Comment:
How do you prevent negative defaults ?

Reply:
I don't prevent them. negative would mean that on the first failure it exceeded the limit. Looked at https://perldoc.perl.org/Getopt/Long.html and the parsing option is `i`nteger. Would you rather negative values cause mtr to error?

Reply:
So in that context there is no difference in output result of negative value and `opt_max_test_fail = 1`, it may be confusing. 
I think it would be better to allow only set of whole numbers (&#x2115;<sub> 0 </sub> )  and for negative values just to raise the error.

Reply:
The prior to this commit case was negative values where ignored, which is just as confusing so point taken, check added.